src/main/java/io/github/fablabsmc/fablabs/impl/fiber/annotation/BackedConfigLeaf.java

```diff
@@ -0,0 +1,173 @@
+import io.github.fablabsmc.fablabs.api.fiber.v1.tree.ConfigBranch;
+import io.github.fablabsmc.fablabs.api.fiber.v1.tree.ConfigLeaf;
+
+/**
+ * A config leaf backed by a field, {@linkplain ConfigType}, and deferred leaf.
+ *
+ * <p>It is used to fetch the backing field's values on each {@link #getValue()} call, to make sure a leaf and its corresponding POJO field are always synchronised.
+ *
+ * @param <R>
+ * @param <S>
+ */
+public class BackedConfigLeaf<R, S> implements ConfigLeaf<S> {
+	final ConfigLeaf<S> backing;
+	final ConfigType<R, S, ?> type;
+	final Object pojo;
+	final Field backingField;
+	private R cachedValue = null;
+	private ConfigBranch parent;
+
+	public BackedConfigLeaf(ConfigLeaf<S> backing, ConfigType<R, S, ?> type, Object pojo, Field backingField) {
+		this.backing = backing;
+		this.type = type;
+		this.pojo = pojo;
+		this.backingField = backingField;
+	}
+
+	@Override
+	public String getName() {
+		return backing.getName();
+	}
+
+	@Override
+	public Map<FiberId, ConfigAttribute<?>> getAttributes() {
+		return backing.getAttributes();
+	}
+
+	@Override
+	public <R, A> Optional<R> getAttributeValue(FiberId id, ConfigType<R, A, ?> type) {
+		return backing.getAttributeValue(id, type);
+	}
+
+	@Override
+	public <A> Optional<A> getAttributeValue(FiberId id, SerializableType<A> expectedType) {
+		return backing.getAttributeValue(id, expectedType);
+	}
+
+	@Override
+	public <A> ConfigAttribute<A> getOrCreateAttribute(FiberId id, SerializableType<A> attributeType, @Nullable A defaultValue) {
+		return backing.getOrCreateAttribute(id, attributeType, defaultValue);
+	}
+
+	@Nullable
+	@Override
+	public ConfigBranch getParent() {
+		return parent;
+	}
+
+	@Override
+	public void attachTo(ConfigBranch parent) throws IllegalTreeStateException {
+		// Attaching/detaching requires careful inspection of whether or not the parent's items contains `this`.
+		// In the case of deferred leaves, the leaf in the parent's items is the deferred leaf and not the backing leaf.
+		// Thus, we can't defer attaching/detaching to the backing leaf, as it will think it's not part of the parent's items.
+		if (this.parent != null && this.parent != parent) {
+			throw new IllegalTreeStateException(this + " needs to be detached before changing the parent");
+		}
+
+		// this node may have already been added by the collection
+		if (parent != null && !parent.getItems().contains(this)) {
+			parent.getItems().add(this);
+		}
+
+		this.parent = parent;
+	}
+
+	@Override
+	public void detach() {
+		// Note: infinite recursion between ConfigNode#detach() and NodeCollection#remove() could occur here,
+		// but the latter performs the actual collection removal before detaching
+		if (this.parent != null) {
+			// here, we also need to avoid triggering a ConcurrentModificationException
+			// thankfully, remove does not cause a CME if it's a no-op
+			this.parent.getItems().remove(this);
+		}
+
+		this.parent = null;
+	}
```

Comment:
Couldn't we delegate those methods to the backing node and thus avoid duplicating the logic ?

Reply:
No: `attachTo` and `detach` will add / remove `this` from the parent. We want the backed node to be added to the collection, and not the backing node.

Reply:
Then we should use inheritance to avoid duplicating the code.

Reply:
`this.parent` does not refer to the same field in both contexts, and we can't use `#getParent` (because we also need the nonexistant `#setParent`).

The duplication of code here is ugly, I agree, but I don't think we can use inheritance here. It wouldn't really fit into the delegation pattern `BackedConfigLeaf` follows anyways.